sympy/functions/elementary/hyperbolic.py

```diff
@@ -336,12 +346,25 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        from sympy import pi
+        arg = self.args[0]
+
+        # `cosh(x)` is real for real and purely imaginary `x`
+        if arg.is_real or arg.is_imaginary:
+            return True
+
+        # cosh(a+ib) = cos(b)*cosh(a) + i*sin(b)*sinh(a)
+        # the imaginary part can be an expression like n*pi
+        # if not, check if the imaginary part is a number
+        re, im = arg.as_real_imag()
+        if (im % pi) == 0:
+            return True
+        elif im.is_number:
+            return False
```

Comment:
I'm worried you aren't using trinary logic correctly.  You should return `True`, `False`, or `None`.  `None` when the answer cannot be determined.  It's better to use `(im % pi).it_zero` than comparing directly with zero.  You can check that for `True`, `False`, and `None` accordingly.